### PR TITLE
rgw: [CORTX-32389] Fix for IAM IPPolicyTest crash

### PR DESCRIPTION
**Pull request by vastradparayya**
problem:
	IAM policy unittest was crashing when we run iam
	policy unittests at IPPolicyTest test suite.
solution:
	In #258 PR we added fix for versionId policy. In
	this fix we added if condition for versionId

Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by sanalpk — APPROVED** [review]

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -901,7 +901,8 @@ void rgw_build_iam_environment(rgw::sal::Store* store,
     s->env.emplace("aws:UserAgent", i->second);
   }
 
-  i = m.find("QUERY_STRING"); {
+  i = m.find("QUERY_STRING"); 
+  if (i != m.end()) {
```
Comment (sachinpunadikar) [inline src/rgw/rgw_op.cc:905]: Add comment stating the reason for checking m.end()

Comment (vastradparayya) [reply src/rgw/rgw_op.cc]: Done

File: src/rgw/rgw_op.cc [hunk 2]
```diff
@@ -901,7 +901,10 @@ void rgw_build_iam_environment(rgw::sal::Store* store,
     s->env.emplace("aws:UserAgent", i->second);
   }
 
-  i = m.find("QUERY_STRING"); {
+  i = m.find("QUERY_STRING");
+  //["QUERY_STRING"] = "", will be always present in the api request even it is empty, 
+  //but for UT this entry may not be there, so adding if check.
```
Comment (sachinpunadikar) [inline src/rgw/rgw_op.cc:906]: comment should say :
// Handle non presence of "QUERY_STRING"

Comment (vastradparayya) [reply src/rgw/rgw_op.cc]: Done

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**